### ISSUE TITLE
Question: Getting sublevel keys from parent levels.

### ISSUE THREAD

**Issue by Industrial** [dominictarr/level-sublevel#100]
Hi.

[This](https://github.com/dominictarr/level-sublevel#example) tells me a sublevel saves the keys so they dont collide with the outer db.

Is it possible for the outer db to get a key from the sub?

Can I get a list of all keys (including sub's, so a tree?) or from a specific point and beneath it?

**Comment by dominictarr** [comment]
No. The outer db cannot see into the sub. in `db = Sublevel(rawdb = level(path))` db is a wrapped sublevel, but rawdb is the unadorned level instance - with `rawdb` you can retrive any key, so if you rebuilt the sublevel key structure you could read them out directly.

What sort of data are you trying to store? can you describe it from the point of view of your application? that would make it much easier for me to understand what you are trying to do

**Comment by Industrial** [comment]
I'm exploring leveldb as a Store component in my React setup. With Flux the React components listen to stores and cause actions which influence the store. There's no more separate Models and Collections as with Backbone for example.

I have been playing with the Redux implementation of Flux but I want to try my own hand at it, possibly using LevelDB as the Store component.

I'm not sure exactly what the use case for sublevels would be but I guess the kind of data I'm trying to store would mostly be results from REST resources and hopefully WebSockets (so also JSON / Objects), maybe even the complete application state so something like:

```js
{
  authentication: {
    token: 'acb123'
  },

  cms: {
    current: { ... },

    list: {
      pagination: { ... },
      items: [ { ... } ]
    }
  }
}
```

I found the query language [jsonquery-engine](https://github.com/eugeneware/jsonquery-engine) and found it pretty cool to be able to use the mongodb query language in the browser so I implemented that too.

One essential thing I'm looking for in the end that would be very awesome would be where a component subscribes to some portion of the store data and just supplies a path e.g. `cms.list.items` so it will receive updates on that part of the tree. If the mongodb query language would support live queries then that would provide very descriptive data dependencies for the components imho but maybe that's a little too much :-)
